tiledb/api/src/datatype/mod.rs

```diff
@@ -529,40 +584,40 @@ macro_rules! fn_typed {
             Datatype::UInt64 => { type $typename = u64; $then },
             Datatype::Float32 => { type $typename = f32; $then },
             Datatype::Float64 => { type $typename = f64; $then },
-            Datatype::Char => unimplemented!(),
-            Datatype::StringAscii => unimplemented!(),
-            Datatype::StringUtf8 => unimplemented!(),
-            Datatype::StringUtf16 => unimplemented!(),
-            Datatype::StringUtf32 => unimplemented!(),
-            Datatype::StringUcs2 => unimplemented!(),
-            Datatype::StringUcs4 => unimplemented!(),
+            Datatype::Char => { type $typename = i8; $then },
```

Comment:
Seems fine for now but I sorta wonder if in the future we should use newtypes for these so that we can do custom display, and etc.

And also now that I'm playing more with the Arrow datatypes I wonder if we should draw some inspiration from them too.

Just musing. Looks good for now

Reply:
Yeah. In an ideal world our type system would get split between concrete and logical types though that's apparently contentious.